src/doom/m_random.c

```diff
@@ -63,3 +63,10 @@ void M_ClearRandom (void)
 {
     rndindex = prndindex = 0;
 }
+
+// inspired by the same routine in Eternity, thanks haleyjd
+int P_SubRandom (void)
+{
+    int r = P_Random();
```

Comment:
How do we prevent compilers from "optimizing" this back to `return P_Random() - P_Random()`? Will adding `const` to `int r` probably help?